ui/app/components/app/network.js

```diff
@@ -2,175 +2,179 @@ import PropTypes from 'prop-types'
 import React, { Component } from 'react'
 
 import classnames from 'classnames'
-import { inherits } from 'util'
 import NetworkDropdownIcon from './dropdowns/components/network-dropdown-icon'
 
-Network.contextTypes = {
-  t: PropTypes.func,
-}
-
-export default Network
-
-inherits(Network, Component)
-
-function Network () {
-  Component.call(this)
-}
-
-Network.prototype.render = function Network () {
-  const context = this.context
-  const networkNumber = this.props.network
-  let providerName, providerNick, providerUrl
-  try {
-    providerName = this.props.provider.type
-    providerNick = this.props.provider.nickname || ''
-    providerUrl = this.props.provider.rpcTarget
-  } catch (e) {
-    providerName = null
+export default class Network extends Component {
+  static contextTypes = {
+    t: PropTypes.func,
   }
-  const providerId = providerNick || providerName || providerUrl || null
-  let iconName
-  let hoverText
 
-  if (providerName === 'mainnet') {
-    hoverText = context.t('mainnet')
-    iconName = 'ethereum-network'
-  } else if (providerName === 'ropsten') {
-    hoverText = context.t('ropsten')
-    iconName = 'ropsten-test-network'
-  } else if (providerName === 'kovan') {
-    hoverText = context.t('kovan')
-    iconName = 'kovan-test-network'
-  } else if (providerName === 'rinkeby') {
-    hoverText = context.t('rinkeby')
-    iconName = 'rinkeby-test-network'
-  } else if (providerName === 'goerli') {
-    hoverText = context.t('goerli')
-    iconName = 'goerli-test-network'
-  } else {
-    hoverText = providerId
-    iconName = 'private-network'
+  static propTypes = {
+    network: PropTypes.number.isRequired,
```

Comment:
```suggestion
    network: PropTypes.string.isRequired,
```

Strangely, the prop referred to as `networkNumber` is in fact a string, not a number.